md-rest-api/src/main/java/com/wipro/ats/bdre/md/rest/ext/AnalyticsAppAPI.java

```diff
@@ -0,0 +1,279 @@
+        for (com.wipro.ats.bdre.md.beans.table.Process process:tableProcessList) {
+            process.setCounter(counter);
+            process.setTableAddTS(DateConverter.dateToString(process.getAddTS()));
+            process.setTableEditTS(DateConverter.dateToString(process.getEditTS()));
+        }
+        restWrapper = new RestWrapper(tableProcessList, RestWrapper.OK);
+        LOGGER.info("Process and Properties for Analytics App process inserted by" + principal.getName());
+
+
+        return restWrapper;
+    }
+
+    @RequestMapping(value = {"/industries"}, method = RequestMethod.GET)
+    @ResponseBody
+    public RestWrapper list(Principal principal) {
+
+        RestWrapper restWrapper = null;
+        try {
+
+            List<AnalyticsApps> analyticsAppsList = new ArrayList<AnalyticsApps>();
+            List<String> industriesList = analyticsAppsDAO.listIndustries();
+            LOGGER.info(industriesList);
+            for (String industry : industriesList) {
+                AnalyticsApps analyticsApp = new AnalyticsApps();
+                analyticsApp.setIndustryName(industry);
+                analyticsAppsList.add(analyticsApp);
+            }
+
+            restWrapper = new RestWrapper(analyticsAppsList, RestWrapper.OK);
+            LOGGER.info("All records listed from Properties by User:" + principal.getName());
+
+        } catch (MetadataException e) {
+            LOGGER.error(e);
+            restWrapper = new RestWrapper(e.getMessage(), RestWrapper.ERROR);
+        }
+        return restWrapper;
+    }
+
+    @RequestMapping(value = {"/category/{industry}"}, method = RequestMethod.GET)
+    @ResponseBody
+    public RestWrapper listCategories(@PathVariable("industry") String industry, Principal principal) {
+
+        RestWrapper restWrapper = null;
+        try {
+
+            List<AnalyticsApps> analyticsAppsList = new ArrayList<AnalyticsApps>();
+            for (String category : analyticsAppsDAO.listCategories(industry)) {
+                AnalyticsApps analyticsApp = new AnalyticsApps();
+                analyticsApp.setIndustryName(industry);
+                analyticsApp.setCategoryName(category);
+                analyticsAppsList.add(analyticsApp);
+            }
+
+            restWrapper = new RestWrapper(analyticsAppsList, RestWrapper.OK);
+            LOGGER.info("All records listed from Properties by User:" + principal.getName());
+
+        } catch (MetadataException e) {
+            LOGGER.error(e);
+            restWrapper = new RestWrapper(e.getMessage(), RestWrapper.ERROR);
+        }
+        return restWrapper;
+    }
+
+    @RequestMapping(value = {"/apps/{industry}/{category}"}, method = RequestMethod.GET)
+    @ResponseBody
+    public RestWrapper listApps(@PathVariable("industry") String industry,@PathVariable("category") String category, Principal principal) {
+
+        RestWrapper restWrapper = null;
+        try {
+            LOGGER.info("Categpry is "+category);
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "Categpry is " 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.openbdre.org/coding_rules#rule_key=squid%3AS1192)